app/javascript/apiService.js

```diff
@@ -107,20 +107,40 @@ const deleteRentalAssistance = async (rentalAssistanceId) => {
   return request.destroy(`/rental-assistances/${rentalAssistanceId}`)
 }
 
-export const createOrUpdateLease = async (leaseToCreateOrUpdate, applicationId) => {
-  const leaseId = leaseToCreateOrUpdate['id']
-  // check if lease_start_date was removed
-  if (!leaseToCreateOrUpdate['lease_start_date']) {
-    leaseToCreateOrUpdate['lease_start_date'] = {}
+const getLeaseRequestData = (rawLeaseObject, primaryApplicantContact) => {
+  return {
+    lease: {
+      ...rawLeaseObject,
+      // TODO: We should consider setting the Tenant on a Lease more explicitly
+      // either via a non-interactable form element or using Salesforce
+      primary_applicant_contact: primaryApplicantContact,
+      // TODO: determine why lease_start_date gets removed
```

Comment:
I think we had an issue where with the way that we were processing the dates with multiple fields that when leasing agents cleared all of the lease start date fields we weren't actually deleting it on Salesforce

Reply:
Do you mind helping me come up with a better way to phrase this comment?

Reply:
How about:
>> TODO: Move removing empty lease start dates to the date component.